packages/playwright-test/src/reporters/azure.ts

```diff
@@ -0,0 +1,481 @@
+  orgUrl: string;
+  projectName: string;
+  logging?: boolean | undefined;
+  isDisabled?: boolean | undefined;
+  environment?: string | undefined;
+  testRunTitle?: string | undefined;
+  uploadAttachments?: boolean | undefined;
+  attachmentsType?: TAttachmentType | undefined;
+}
+
+interface TestResultsToTestRun {
+  statusCode: number;
+  result: Result;
+  headers: Headers;
+}
+interface Result {
+  count: number;
+  value?: ValueEntity[] | null;
+}
+interface ValueEntity {
+  id: number;
+  project: Project;
+  outcome: string;
+  testRun: TestRun;
+  priority: number;
+  url: string;
+  lastUpdatedBy: LastUpdatedBy;
+}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface Project {
+}
+interface TestRun {
+  id: string;
+}
+interface LastUpdatedBy {
+  displayName?: null;
+  id?: null;
+}
+interface Headers {
+  'cache-control': string;
+  'pragma': string;
+  'content-length': string;
+  'content-type': string;
+  'expires': string;
+  'p3p': string;
+  'x-tfs-processid': string;
+  'strict-transport-security': string;
+  'activityid': string;
+  'x-tfs-session': string;
+  'x-vss-e2eid': string;
+  'x-vss-senderdeploymentid': string;
+  'x-vss-userdata': string;
+  'x-frame-options': string;
+  'request-context': string;
+  'access-control-expose-headers': string;
+  'x-content-type-options': string;
+  'x-cache': string;
+  'x-msedge-ref': string;
+  'date': string;
+  'connection': string;
+}
+
+class AzureDevOpsReporter implements Reporter {
+  private testApi!: Test.ITestApi;
+  private coreApi!: ICoreApi;
+  private _options: AzureReporterOptions;
+  private publishedResultsCount = 0;
+  private resultsToBePublished: string[] = [];
+  private connection!: WebApi;
+  private orgUrl!: string;
+  private projectName!: string;
+  private environment?: string;
+  private planId = 0;
+  private runId!: number;
+  private logging = true;
+  private isDisabled = false;
+  private testRunTitle = '';
+  private uploadAttachments = false;
+  private attachmentsType?: TAttachmentType;
+  private token: string = '';
+
+  public constructor(options: AzureReporterOptions = {} as AzureReporterOptions) {
+    this._options = options;
+
+    this.orgUrl = this._options.orgUrl;
+    this.projectName = this._options.projectName;
+    this.environment = this._options.environment || undefined;
+    this.planId = this._options.planId;
+    this.logging = (typeof this._options.logging === 'undefined') ? true : this._options.logging;
+    this.isDisabled = this._options.isDisabled || false;
+    this.testRunTitle = `${this.environment ? `[${this.environment}]:` : ''} ${this._options.testRunTitle || 'Playwright Test Run'}` ||
+      `${this.environment ? `[${this.environment}]:` : ''}Test plan ${this.planId}`;
+    this.uploadAttachments = this._options.uploadAttachments || false;
+    this.attachmentsType = this._options.attachmentsType;
+    this.token = this._options.token;
+
+    if (!this.orgUrl || this.orgUrl.length === 0) {
```

Comment:
```suggestion
    if (!this.orgUrl) {
```
an empty string is always falsy, so here and below you can just do !foo.